### PR TITLE
Add ttl to datasync mongo

### PR DESCRIPTION
**Pull request by ssd71**
This PR adds TTL for datasync mongo, It is a work in progress, related to #1472

### CODE REVIEW

**Comment by ssd71** [comment]
For implementing this, I am using [TTL Indexes](https://docs.mongodb.com/manual/core/index-ttl/) from MongoDB, I think while this is not available for Postgres, It is the most optimal approach for MongoDB.

I was confused if we should use cron for this: If there are multiple instances of Graphback running, we might run into issues with scaling, although to be fair, if we might use a filter to delete documents(maybe `_ttl < Date.now()`?), I don't see many issues with this

WDYT

**Comment by wtrocki** [comment]
Now I get `concurency`. The way I seen it is that we have one job starting once some time and it it checks TTL value on deleted fields and prune them. TTL indexes are more sophisticated mechanism. Any benefits of that?

**Comment by wtrocki** [comment]
So we need job that does something - and we will get other kinds of jobs in future that TTL indexes will not support

**Comment by ssd71** [comment]
I see what you mean, thanks :D

File: docs/datasync/datasync-soft-delete.md [hunk 1]
```diff
@@ -61,22 +68,17 @@ type CommentDeltaList {
...
   lastSync: String
 }
-
-type Query {
-  ...
-  syncComments(lastSync: String!, filter: CommentFilter): CommentDeltaList!
-}
 ```
 
- The `@delta` annotation adds a `sync` query or a delta query as shown above. This allows you to get all the changed(updates and deletes) documents in a collection since the `lastSync` timestamp. Internally this uses the `updatedAt` database field to check if any documents in the database have been modified, by comparing client provided `lastSync` timestamp value. The documents to be returned in a `sync` query may also be filtered, so you can, for example, only `sync` comments on a specific post.
```
Comment (wtrocki) [inline docs/datasync/datasync-soft-delete.md:71]: I will have 

# Advanced  

chapter and then mention delta there.

**Comment by ssd71** [comment]
Closing this for now as new approach needed, will open new PR with forementioned approach

**Comment by wtrocki** [comment]
@ssd71 I would rather keep the code and PR for context but it is ok for you to manage things you want to work.

**Pull request closed without merging**